### PR TITLE
Scroll to Top Button addition

### PR DESCRIPTION
**Pull request by harsh07bharvada**
## Proposed Changes

  - Added a ScrollToTopButton Component
  - Added the styles for the same
  - Formatted the code

### CODE REVIEW

File: src/assets/up.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#a5b2bf" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-arrow-up"><line x1="12" y1="19" x2="12" y2="5"></line><polyline points="5 12 12 5 19 12"></polyline></svg>
```
Comment (diogo405) [inline src/assets/up.svg:1]: Please add a new line at the end of the file :)

File: src/components/Container.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-import Results from './Results.js'
-import data from '../json/data.json'
-import './Container.css'
+import { useState, useEffect } from "react";
+import Header from "./Header.js";
+import Search from "./Search.js";
+import Results from "./Results.js";
+import ScrollToTopButton from "./ScrollToTopButton";
+import data from "../json/data.json";
+import "./Container.css";
```
Comment (diogo405) [inline src/components/Container.js:7]: It'd be nice to sticky with single quotes and no semi colon.

File: src/components/Container.js [hunk 3]
```diff
@@ -12,49 +13,48 @@ function Container() {
...
-    let newEmojis = []
+    let newEmojis = [];
 
     let queryLength = queryKeywords.length;
 
     let queryLengthSum = 0;
 
-    console.log("\n\n\n NEW LINE \n\n\n")
-
+    console.log("\n\n\n NEW LINE \n\n\n");
```
Comment (diogo405) [inline src/components/Container.js:28]: We can remove this.

File: src/components/ScrollToTopButton.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  };
+
+  useEffect(() => {
+    const toggleVisibility = () => {
+      if (window.pageYOffset > 500) {
+        setScrollToTop(true);
+      } else {
+        setScrollToTop(false);
+      }
+    };
```
Comment (diogo405) [inline src/components/ScrollToTopButton.js:22]:
```suggestion
    const toggleVisibility = () => {
      setScrollToTop(window.pageYOffset > 500);
    };
```

File: src/components/ScrollToTopButton.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    window.addEventListener("scroll", toggleVisibility);
+
+    return () => window.removeEventListener("scroll", toggleVisibility);
```
Comment (diogo405) [inline src/components/ScrollToTopButton.js:26]: Single quotes for string as well :)

**Review by diogo405 — CHANGES_REQUESTED** [review]

**Comment by harsh07bharvada** [comment]
Thanks for the change request! Will do the same.

**Comment by harsh07bharvada** [comment]
I cannot see the change request for particular lines. Can you please add comments for the specific lines?

File: src/components/ScrollToTopButton.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      setScrollToTop(window.pageYOffset > 500);
+    };
+
+    window.addEventListener('scroll', toggleVisibility);
+
+    return () => window.removeEventListener('scroll', toggleVisibility);
+  }, []);
+
+  return (
+    <div className='scroll-to-top-wrapper'>
```
Comment (diogo405) [inline src/components/ScrollToTopButton.js:26]: Here and all JSX we can keep double quotes sorry.

**Comment by diogo405** [comment]
> I cannot see the change request for particular lines. Can you please add comments for the specific lines?

Yeah weird, might have done something wrong sorry. I've updated with the comment to specific lines.

File: src/components/Container.js [hunk 7]
```diff
@@ -53,6 +54,7 @@ function Container() {
       <Header />
       <Search onChange={onChange} />
       <Results emojiFiltered={searchQuery === "" ? emojiData : newEmojiData} />
+      <ScrollToTopButton/>
```
Comment (diogo405) [inline src/components/Container.js:57]:
```suggestion
      <ScrollToTopButton />
```

**Review by diogo405 — APPROVED** [review]

**Pull request merged**